### ISSUE TITLE
0.17.0-beta.13: 'Module npm:moment@2.12.0/moment.js not present.' when building

### ISSUE THREAD

**Issue by adamburgess** [jspm/jspm-cli#1715]
occurs in beta 12 and 13. (would've reported earlier but was blocked on #1599 so hadn't upgraded yet)

steps to reproduce:

`npm init -y && npm install jspm@beta && jspm init -y && jspm install moment && jspm build moment --skip-encode-names && node build`

output:
```
(function (exports, require, module, __filename, __dirname) { !function(e){function r(e,r,o){return 4===arguments.length?t.apply(this,arguments):void n(e,{declarative:!0,deps:r,declare:o})}function t(e,r,t,o){n(e,{declarative:!1,deps:r,executingRequire:t,execute:o})}function n(e,r){r.name=e,e in p||(p[e]=r),r.normalizedDeps=r.deps}function o(e,r){if(r[e.groupIndex]=r[e.groupIndex]||[],-1==v.call(r[e.groupIndex],e)){r[e.groupIndex].push(e);for(var t=0,n=e.normalizedDeps.length;n>t;t++){var a=e.normalizedDeps[t],u=p[a];if(u&&!u.evaluated){var d=e.groupIndex+(u.declarative!=e.declarative);if(void 0===u.groupIndex||u.groupIndex<d){if(void 0!==u.groupIndex&&(r[u.groupIndex].splice(v.call(r[u.groupIndex],u),1),0==r[u.groupIndex].length))throw new TypeError("Mixed dependency cycle detected");u.groupIndex=d}o(u,r)}}}}function a(e){var r=p[e];r.groupIndex=0;var t=[];o(r,t);for(var n=!!r.declarative==t.length%2,a=t.length-1;a>=0;a--){for(var u=t[a],i=0;i<u.length;i++){var s=u[i];n?d(s):l(s)}
 Module npm:moment@2.12.0/moment.js not present.
```

notes:
without `--skip-encode-names` it'll print `Module 1 not present.`
also occurs the exact same way with bluebird: `jspm install bluebird && jspm build bluebird && node build`

**Comment by JustinWilabyPrecor** [comment]
This could be a moment issue.  Moment's `package.json` shows `"main": "./moment.js"` note the relative path.  getting rid of the dot whack and reinstalling via jspm should solve the problem.

**Comment by AdamBurgess** [comment]
If the relative path works in npm, it should work in JSPM. Regardless, it worked in the previous beta.

**Comment by JustinWilabyPrecor** [comment]
Try removing the dot whack in `moment@2.12.0.json` and build again.

**Comment by guybedford** [comment]
Thanks, I've added a fix for this in https://github.com/systemjs/builder/commit/06f6c3fc3e8e3e9fbaf383619a66f032b801893d.

**Comment by AdamBurgess** [comment]
Ok, maybe bluebird's error was different then?
with your patch:

```
$ jspm build bluebird --skip-encode-names
     Creating the single-file build for bluebird...

       npm:bluebird@3.3.5/js/browser/bluebird.js

ok   Built into build.js with source maps, unminified as umd.
```

```
$ node build
(function (exports, require, module, __filename, __dirname) { *snip*
 Module npm:bluebird@3.3.5/#:./js/release/bluebird.js not present.
```

note the `js/release/bluebird.js` is the entrypoint for node but not browser builds. Still, building it for node gives the same outcome:
```
$ jspm build bluebird --node --skip-encode-names
     Creating the single-file build for bluebird...

       npm:bluebird@3.3.5/js/release/any.js
       npm:bluebird@3.3.5/js/release/async.js
       npm:bluebird@3.3.5/js/release/bind.js
       npm:bluebird@3.3.5/js/release/bluebird.js
       npm:bluebird@3.3.5/js/release/call_get.js
       npm:bluebird@3.3.5/js/release/cancel.js
       npm:bluebird@3.3.5/js/release/catch_filter.js
       npm:bluebird@3.3.5/js/release/context.js
       npm:bluebird@3.3.5/js/release/debuggability.js
       npm:bluebird@3.3.5/js/release/direct_resolve.js
       npm:bluebird@3.3.5/js/release/each.js
       npm:bluebird@3.3.5/js/release/errors.js
       npm:bluebird@3.3.5/js/release/es5.js
       npm:bluebird@3.3.5/js/release/filter.js
       npm:bluebird@3.3.5/js/release/finally.js
       npm:bluebird@3.3.5/js/release/generators.js
       npm:bluebird@3.3.5/js/release/join.js
       npm:bluebird@3.3.5/js/release/map.js
       npm:bluebird@3.3.5/js/release/method.js
       npm:bluebird@3.3.5/js/release/nodeback.js
       npm:bluebird@3.3.5/js/release/nodeify.js
       npm:bluebird@3.3.5/js/release/promise.js
       npm:bluebird@3.3.5/js/release/promise_array.js
       npm:bluebird@3.3.5/js/release/promisify.js
       npm:bluebird@3.3.5/js/release/props.js
       npm:bluebird@3.3.5/js/release/queue.js
       npm:bluebird@3.3.5/js/release/race.js
       npm:bluebird@3.3.5/js/release/reduce.js
       npm:bluebird@3.3.5/js/release/schedule.js
       npm:bluebird@3.3.5/js/release/settle.js
       npm:bluebird@3.3.5/js/release/some.js
       npm:bluebird@3.3.5/js/release/synchronous_inspection.js
       npm:bluebird@3.3.5/js/release/thenables.js
       npm:bluebird@3.3.5/js/release/timers.js
       npm:bluebird@3.3.5/js/release/using.js
       npm:bluebird@3.3.5/js/release/util.js

ok   Built into build.js with source maps, unminified as umd.
```

```
$ node build
(function (exports, require, module, __filename, __dirname) { *snip*
 Module npm:bluebird@3.3.5/#:./js/release/bluebird.js not present.
```

A simple test like `import bb from 'bluebird'; export default bb` builds and runs fine, which is peculiar.
I encountered this bug wanting to do something like `jspm build bluebird --node --format global --global-name Promise` and inlining it at the top of my builds.

**Comment by guybedford** [comment]
Thanks, yes this is another bug... I'll look into it.

**Comment by guybedford** [comment]
Fixed in https://github.com/systemjs/builder/commit/104ff4f52f92ee1ade3c8ef91e9207e58c337f95.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.14.
